x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -212,6 +213,14 @@ public ApiKeyService(Settings settings, Clock clock, Client client, XPackLicense
             this.apiKeyAuthCache = CacheBuilder.<String, ListenableFuture<CachedApiKeyHashResult>>builder()
                 .setExpireAfterAccess(ttl)
                 .setMaximumWeight(maximumWeight)
+                .removalListener(notification -> {
+                    if (RemovalNotification.RemovalReason.EVICTED == notification.getRemovalReason()) {
+                        if (getApiKeyAuthCache().count() >= maximumWeight) {
```

Comment:
This is a reasonable approximation to limit the logging to "eviction due to cache size, not ttl".